backend/main/shared/notification-service.go

```diff
@@ -0,0 +1,175 @@
+package shared
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
+
+type LeanplumBatchBody struct {
+	AppId      string              `json:"appId"`
+	ClientKey  string              `json:"clientKey"`
+	ApiVersion string              `json:"apiVersion"`
+	Time       int64               `json:"time"`
+	Data       []LeanplumBatchData `json:"data"`
+}
+
+type LeanplumBatchData struct {
+	Action    string              `json:"action"`
+	Time      int64               `json:"time"`
+	UserId    string              `json:"userId"`
+	MessageId string              `json:"messageId"`
+	Values    LeanplumEmailValues `json:"values"`
+}
+
+type LeanplumEmailValues struct {
+	CommunityName     string `json:"CommunityName"`
+	ProposalTitle     string `json:"ProposalTitle"`
+	CommunityImageUrl string `json:"CommunityImageUrl"`
+	EndTime           string `json:"EndTime"`
+	ProposalUrl       string `json:"ProposalUrl"`
+}
+
+func sendProposalEmails(p EmailTaskPayload) {
+	var fileResponse map[string]interface{}
+	jobId := getExportJobId(p.CommunityId)
+	for {
+		fileResponse = getExportFiles(jobId)
+		if fileResponse["state"] == "RUNNING" {
+			time.Sleep(5 * time.Second)
+		} else {
+			log.Info().Msgf("LP Export Job Complete with Response: %+v", fileResponse)
+			break
+		}
+	}
+	fileArr := fileResponse["files"].([]interface{})
```

Comment:
not a huge fan of casting this way but the leanplum api can return different response object shapes depending on the result, so it was more challenging to define response type structs